### ISSUE TITLE
Migrations for new schemas leave MV's in unusable state

### ISSUE THREAD

**Issue by whatnick** [opendatacube/datacube-explorer#177]
Successfully performed migration to new schema using `cubedash-gen --init` . However Materialized View are created with no data and are not usable by application code. The following error is reported in logs:

```
HINT:  Use the REFRESH MATERIALIZED VIEW command.
--
  |   | 2020-08-15 09:40:07 | psycopg2.errors.ObjectNotInPrerequisiteState: materialized view "mv_region" has not been populated
```

Running 
```
REFRESH MATERIALIZED VIEW cubedash.mv_region;
```

From psql helps. The migration scripts for the application need some work before release.

However some invalid geometries may block this :
```
nci_20200621=> REFRESH MATERIALIZED VIEW cubedash.mv_region;

ERROR:  GEOSUnaryUnion: TopologyException: Input geom 1 is invalid: Self-intersection at or near point 177.74431629145585 -70.041615841146097 at 177.74431629145585 -70.041615841146097
```

**Comment by jeremyh** [comment]
The view is populated at the end of the generate command (unless `--no-refresh-stats` is given)

... I'm guessing this error was either transient (while the migration was running, but before the last step), or the refresh failed due to the invalid geometry like you mentioned. The logs might help (does it mention "refreshing statistics", and were there errors? -- hopefully it returned error code if so).

We definitely should update the view so that a single bad geometry doesn't block everything -- related to #172

**Comment by whatnick** [comment]
Here is the stacktrace :
```
'Refreshing statistics...Traceback (most recent call last):\n'
'  File "/env/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 1278, in _execute_context\n'
'    cursor, statement, parameters, context\n'
'  File "/env/lib/python3.6/site-packages/sqlalchemy/engine/default.py", line 593, in do_execute\n'
'    cursor.execute(statement, parameters)\n'
'psycopg2.errors.InternalError_: GEOSUnaryUnion: TopologyException: Input geom 1 is invalid: Self-intersection at or near point 177.74431629145585 -70.041615841146097 at 177.74431629145585 -70.041615841146097\n'
'\n'
'\n'
'The above exception was the direct cause of the following exception:\n'
'\n'
'Traceback (most recent call last):\n'
'  File "/env/bin/cubedash-gen", line 8, in <module>\n'
'    sys.exit(cli())\n'
'  File "/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__\n'
'    return self.main(*args, **kwargs)\n'
'  File "/env/lib/python3.6/site-packages/click/core.py", line 782, in main\n'
'    rv = self.invoke(ctx)\n'
'  File "/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke\n'
'    return ctx.invoke(self.callback, **ctx.params)\n'
'  File "/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke\n'
'    return callback(*args, **kwargs)\n'
'  File "/env/lib/python3.6/site-packages/datacube/ui/click.py", line 196, in new_func\n'
'    return f(parsed_config, *args, **kwargs)\n'
'  File "/env/lib/python3.6/site-packages/cubedash/generate.py", line 369, in cli\n'
'    store.refresh_stats(concurrently=force_concurrently)\n'
'  File "/env/lib/python3.6/site-packages/cubedash/summary/_stores.py", line 233, in refresh_stats\n'
'    refresh_supporting_views(self._engine, concurrently=concurrently)\n'
'  File "/env/lib/python3.6/site-packages/cubedash/summary/_schema.py", line 370, in refresh_supporting_views\n'
'    """\n'
'  File "/env/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 2238, in execute\n'
'    return connection.execute(statement, *multiparams, **params)\n'
'  File "/env/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 1006, in execute\n'
'    return self._execute_text(object_, multiparams, params)\n'
'  File "/env/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 1181, in _execute_text\n'
'    parameters,\n'
'  File "/env/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 1318, in _execute_context\n'
'    e, statement, parameters, cursor, context\n'
'  File "/env/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 1512, in _handle_dbapi_exception\n'
'    sqlalchemy_exception, with_traceback=exc_info[2], from_=e\n'
'  File "/env/lib/python3.6/site-packages/sqlalchemy/util/compat.py", line 178, in raise_\n'
'    raise exception\n'
'  File "/env/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 1278, in _execute_context\n'
'    cursor, statement, parameters, context\n'
'  File "/env/lib/python3.6/site-packages/sqlalchemy/engine/default.py", line 593, in do_execute\n'
'    cursor.execute(statement, parameters)\n'
'sqlalchemy.exc.InternalError: (psycopg2.errors.InternalError_) GEOSUnaryUnion: TopologyException: Input geom 1 is invalid: Self-intersection at or near point 177.74431629145585 -70.041615841146097 at 177.74431629145585 -70.041615841146097\n'
'\n'
'[SQL: \n'
'    refresh materialized view  cubedash.mv_region;\n'
'    ]\n'
'(Background on this error at: http://sqlalche.me/e/13/2j85)\n'
```

**Comment by jeremyh** [comment]
Fixed by #180 as it's no longer a materialised view. Geometries are also being buffered now on union.
